### DISCUSSION TITLE
Deployment error a2.defineComponent is not a function on cloudflare pages and workers

### DISCUSSION THREAD

**Discussion by EmilyYond** [nitrojs/nitro#1391] [Q&A]
Hoping someone can point me in the right direction!

I'm trying to deploy my site and have the following setup:

Nuxt 3 in SSR mode

Package.json:
```
"devDependencies": {
    "@nuxt/devtools": "latest",
    "@nuxt/image": "^1.0.0-rc.1",
    "@types/node": "^18",
    "autoprefixer": "^10.4.14",
    "nuxt": "^3.6.0",
    "postcss": "^8.4.24",
    "prisma": "^4.16.1",
    "sass": "^1.63.6",
    "sass-loader": "^13.3.2",
    "tailwindcss": "^3.3.2",
    "typescript": "latest",
  },
  "dependencies": {
    "@fawmi/vue-google-maps": "0.9.72",
    "@nuxtjs/strapi": "^1.9.3",
    "axios": "^1.4.0",
    "dayjs": "^1.11.8",
    "dotenv": "^16.3.1",
    "node-ical": "^0.16.1",
    "pinia": "^2.1.4",
    "postcode": "^5.1.0",
    "swiper": "^10.0.2",
    "v-calendar": "^2.4.1",
    "vite-svg-loader": "^4.0.0",
    "vue-imask": "^7.0.1",
    "vue-social-sharing": "^3.0.9",
    "vue3-tel-input": "^1.0.4"
  }
```

Wrangler.toml:
```
main = "./.output/server/index.mjs"
workers_dev = true
compatibility_date = "2022-09-01"
account_id = "***"
[site]
bucket = ".output/public"
```

I have also tried using pages with the default build settings and hit the same error during the deploy/publish to cloudflare stage, as below:

![Screenshot 2023-07-04 at 10 15 25](https://github.com/unjs/nitro/assets/128067060/04044a92-f199-4bf4-b3ed-51f65d4b10a3)

I have followed https://nitro.unjs.io/deploy/providers/cloudflare but keep hitting this same error and I don't know where to even begin debugging this. Does anyone have any ideas?

**Comment by Hebilicious** [comment]
Hi @EmilyYond , can you please open an issue with a reproduction in the Nuxt repo ?

**Reply by EmilyYond** [reply]
@Hebilicious done https://github.com/nuxt/nuxt/issues/21938
